### PR TITLE
Use ArgumentException.ThrowIfNullOrEmpty() in more places

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Use ArgumentException.ThrowIfNullOrEmpty(); where possible

## PR Context
Contributes to #19212  

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by xtqqczze** [comment]
@daxian-dbw Can we make changes to `Microsoft.Management.UI.Internal`. Notably, #18784 made changes there.

Previous guidance has been the module is frozen, but we have not documented this in the repository.

**Comment by daxian-dbw** [comment]
For `Microsoft.Management.UI.Internal`, we ship it and own the code, so I think it's OK to make changes to it.
But for code that we don't build/ship (e.g. ScheduledJob, LocalAccounts), we should avoid changing it.
For the code that comes from .NET (e.g. ComInterop), we should avoid changing it unless the original code in dotnet/runtime was also updated to use `ArgumentException.ThrowIfNullOrEmpty`.

**Comment by xtqqczze** [comment]
@CarloToso I see you have a few pull requests that have been open for some time. In the case of this PR, although it is marked as _Small_ , the changes cover multiple modules, so it may take reviewers some time to review and approve the changes. If you would like, I can use `git cherry-pick` to split your contribution (preserving attribution).

**Comment by xtqqczze** [comment]
@CarloToso I could also document which changes are affected by the PowerShell [breaking changes contract](https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/breaking-change-contract.md).

**Comment by CarloToso** [comment]
@xtqqczze Thank you very much, I'll gladly accept your help 😄

File: src/System.Management.Automation/utils/perfCounters/PSPerfCountersMgr.cs [hunk 1]
```diff
@@ -233,8 +233,7 @@ public bool AddCounterSetInstance(CounterSetRegistrarBase counterSetRegistrarIns
         {
             if (counterSetName == null)
             {
-                ArgumentNullException argNullException =
-                    new ArgumentNullException("counterSetName");
+                ArgumentNullException argNullException = new ArgumentNullException("counterSetName");
```
Comment (iSazonov) [inline src/System.Management.Automation/utils/perfCounters/PSPerfCountersMgr.cs:236]: We could use `nameof` here and below.

**Comment by iSazonov** [comment]
> For `Microsoft.Management.UI.Internal`, we ship it and own the code, so I think it's OK to make changes to it. But
> 
> * for code that we don't build/ship (e.g. `ScheduledJob`, `LocalAccounts`), we should avoid changing it.
> * for the code that came from .NET (e.g. `ComInterop`), we should avoid changing it unless the original code in dotnet/runtime was also updated to use `ArgumentException.ThrowIfNullOrEmpty`.

Also WinRM code. Very sensitive.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs [hunk 2]
```diff
@@ -419,10 +419,7 @@ public static bool TryExecute(this RoutedCommand command, object parameter, IInp
         {
             ArgumentNullException.ThrowIfNull(templateParent);
```
Comment (daxian-dbw) [inline src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs:421]: Please remove this empty line.

File: src/Microsoft.WSMan.Management/CurrentConfigurations.cs [hunk 3]
```diff
@@ -146,10 +140,7 @@ public void UpdateOneConfiguration(string pathToNodeFromRoot, string configurati
         {
             ArgumentNullException.ThrowIfNull(pathToNodeFromRoot);
```
Comment (daxian-dbw) [inline src/Microsoft.WSMan.Management/CurrentConfigurations.cs:142]: Please remove this empty line.

File: src/Microsoft.Management.UI.Internal/ManagementList/FilterProviders/FilterRulePanelItem.cs [hunk 4]
```diff
@@ -81,10 +81,7 @@ public FilterRulePanelItem(FilterRule rule, string groupId)
         {
             ArgumentNullException.ThrowIfNull(rule);
```
Comment (daxian-dbw) [inline src/Microsoft.Management.UI.Internal/ManagementList/FilterProviders/FilterRulePanelItem.cs:83]: Please remove this empty line.

File: src/Microsoft.WSMan.Management/WsManHelper.cs [hunk 5]
```diff
@@ -180,10 +180,7 @@ internal string GetResourceMsgFromResourcetext(string rscname)
         {
             ArgumentNullException.ThrowIfNull(resourceManager);
```
Comment (daxian-dbw) [inline src/Microsoft.WSMan.Management/WsManHelper.cs:182]: Please remove this empty line.

File: src/System.Management.Automation/cimSupport/cmdletization/ObjectModelWrapper.cs [hunk 6]
```diff
@@ -19,14 +19,10 @@ internal void Initialize(PSCmdlet cmdlet, string className, string classVersion,
         {
             ArgumentNullException.ThrowIfNull(cmdlet);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/cimSupport/cmdletization/ObjectModelWrapper.cs:21]: Please remove this empty line.

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs (hunk 2 above)
Comment (CarloToso) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs:421]: Done 👍

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Review by daxian-dbw — APPROVED** [review]

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs [hunk 7]
```diff
@@ -419,10 +419,7 @@ public static bool TryExecute(this RoutedCommand command, object parameter, IInp
...
 
-            if (string.IsNullOrEmpty(childName))
-            {
-                throw new ArgumentNullException("childName");
-            }
+            ArgumentException.ThrowIfNullOrEmpty(childName);
```
Comment (iSazonov) [inline src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs:422]: Extra blank line 😄

Comment (daxian-dbw) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs:422]: @CarloToso you and I both missed this one ...

Comment (CarloToso) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs:422]: 😅 (I fixed it in #19241)

Comment (daxian-dbw) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/WpfHelp.cs:422]: Thank you! 😄

**Pull request merged**